overlord/snapstate/handlers.go

```diff
@@ -1942,6 +1942,128 @@ func (m *SnapManager) doPreferAliases(t *state.Task, _ *tomb.Tomb) error {
 	return nil
 }
 
+// nearlyReady returns whether all lanes (other than me) are Ready.
+func nearlyReady(me string, change *state.Change) bool {
+	for _, task := range change.Tasks() {
+		if me == task.ID() {
+			// ignore self
+			continue
+		}
+		if !task.Status().Ready() {
+			return false
+		}
+	}
+	return true
+}
+
+// laneSnaps returns the instance names from the first SnapSetup in each
+// non-zero Done lane in the refresh batch leading up the given task
+func laneSnaps(reTask *state.Task) []string {
```

Comment:
The description sounds very specific - it sounds like the task must be part of a refresh batch - should this code check that the task actually belongs to a change that is about refresh?

Reply:
It's not _that_ specific -- anything that uses a SnapSetup would return the right info. I'll adjust the description.

Reply:
eh, the description is now unclear. Trying another tack.